src/librustc/middle/resolve_lifetime.rs

```diff
@@ -992,11 +992,14 @@ impl<'a, 'tcx> LifetimeContext<'a, 'tcx> {
             let unsubst = if let Some(id) = self.hir_map.as_local_node_id(def_id) {
                 &map.object_lifetime_defaults[&id]
             } else {
-                let cstore = &self.sess.cstore;
                 self.xcrate_object_lifetime_defaults.entry(def_id).or_insert_with(|| {
-                    cstore.item_generics_cloned(def_id).types.into_iter().map(|def| {
-                        def.object_lifetime_default
-                    }).collect()
+                    ty::tls::with_opt(|opt_tcx| {
```

Comment:
@nikomatsakis it seems that this call isn't able to retrieve `tcx` properly in a few scenarios. Is there a more appropriate way to reach `tcx` from this location?